File: src/test/java/komu/blunt/types/KindTest.java
package komu.blunt.types;

import org.junit.Test;

import static komu.blunt.types.Kind.STAR;
import static komu.blunt.types.Kind.arrow;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class KindTest {

    @Test
    public void stringRepresentationOfKinds() {
        assertThat(STAR.toString(), is("*"));
        
        assertThat(arrow(STAR, STAR).toString(), is("* -> *"));
        
        assertThat(arrow(STAR, arrow(STAR, STAR)).toString(), is("* -> * -> *"));
        assertThat(arrow(STAR, arrow(STAR, arrow(STAR, STAR))).toString(), is("* -> * -> * -> *"));

        assertThat(arrow(arrow(STAR, STAR), STAR).toString(), is("(* -> *) -> *"));
    }
}

File: src/test/kotlin/komu/blunt/types/KindTest.kt
package komu.blunt.types

import komu.blunt.types.Kind.Arrow
import komu.blunt.types.Kind.Star
import org.junit.Test
import kotlin.test.assertEquals

class KindTest {

    @Test
    fun stringRepresentationOfKinds() {
        assertEquals("*", Star.toString());
        assertEquals("* -> *", Arrow(Star, Star).toString());
        assertEquals("* -> * -> *", Arrow(Star, Arrow(Star, Star)).toString());
        assertEquals("* -> * -> * -> *", Arrow(Star, Arrow(Star, Arrow(Star, Star))).toString());
        assertEquals("(* -> *) -> *", Arrow(Arrow(Star, Star), Star).toString());
    }
}
